In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

# to display all columns
pd.pandas.set_option('display.max_columns', None)

In [2]:
df=pd.read_csv('Downloads/FoodBalanceSheets_E_Africa_NOFLAG (1).csv', encoding='Latin-1')
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [3]:
# Question 4
df1=df[df['Area']=='Madagascar'].groupby(['Element','Area']).sum()
df1

Area Code  Item Code  \
Element                                Area                               
Domestic supply quantity               Madagascar      15351     318888   
Export Quantity                        Madagascar      13416     279445   
Fat supply quantity (g/capita/day)     Madagascar      14319     299405   
Feed                                   Madagascar       4515      94030   
Food                                   Madagascar      14190     295817   
Food supply (kcal/capita/day)          Madagascar      14319     299405   
Food supply quantity (kg/capita/yr)    Madagascar      14190     295817   
Import Quantity                        Madagascar      14319     297945   
Losses                                 Madagascar       7095     148102   
Other uses (non-food)                  Madagascar       6579     138246   
Processing                             Madagascar       9804     201929   
Production                             Madagascar      14190     295071   
Protein supply quantity (g/capita/day) Madagascar      14319     299405   
Residuals                              Madagascar      13545     280011   
Seed                                   Madagascar       2709      56067   
Stock Variation                        Madagascar      12642     262174   
Total Population - Both sexes          Madagascar        129       2501   

                                                   Element Code     Y2014  \
Element                                Area                                 
Domestic supply quantity               Madagascar        630819  31578.76   
Export Quantity                        Madagascar        614744    425.48   
Fat supply quantity (g/capita/day)     Madagascar         75924     88.72   
Feed                                   Madagascar        193235   2178.76   
Food                                   Madagascar        565620  21405.42   
Food supply (kcal/capita/day)          Madagascar         73704   7945.00   
Food supply quantity (kg/capita/yr)    Madagascar         70950    907.59   
Import Quantity                        Madagascar        622821   2010.71   
Losses                                 Madagascar        281765   2252.00   
Other uses (non-food)                  Madagascar        262854   4348.58   
Processing                             Madagascar        389956   1262.00   
Production                             Madagascar        606210  30337.51   
Protein supply quantity (g/capita/day) Madagascar         74814    177.79   
Residuals                              Madagascar        542850   -603.00   
Seed                                   Madagascar        116067    728.00   
Stock Variation                        Madagascar        497056    352.00   
Total Population - Both sexes          Madagascar           511  23590.00   

                                                      Y2015     Y2016  \
Element                                Area                             
Domestic supply quantity               Madagascar  31214.98  31732.08   
Export Quantity                        Madagascar    494.75    535.06   
Fat supply quantity (g/capita/day)     Madagascar     91.85     95.33   
Feed                                   Madagascar   2070.22   2121.40   
Food                                   Madagascar  21120.65  21435.61   
Food supply (kcal/capita/day)          Madagascar   7685.00   7669.00   
Food supply quantity (kg/capita/yr)    Madagascar    871.59    861.17   
Import Quantity                        Madagascar   1721.80   1901.47   
Losses                                 Madagascar   2176.00   2215.00   
Other uses (non-food)                  Madagascar   3957.12   4114.08   
Processing                             Madagascar   1803.00   1770.00   
Production                             Madagascar  29482.89  29889.65   
Protein supply quantity (g/capita/day) Madagascar    173.05    173.40   
Residuals                              Madagascar   -616.00   -642

In [4]:
# Question 4 contd
df1.loc['Protein supply quantity (g/capita/day)',['Y2015']]

Y2015
Area              
Madagascar  173.05

In [5]:
# Question 7
df2=df.groupby(['Area'])['Y2017'].sum().sort_values(ascending=False)
df2

Area
Nigeria                        1483268.23
Egypt                           866379.92
South Africa                    517590.54
Ethiopia                        448683.76
Morocco                         388495.36
Ghana                           337599.06
Algeria                         325644.27
United Republic of Tanzania     322616.85
Kenya                           264660.66
Sudan                           239931.92
Cameroon                        232030.43
Angola                          229159.57
Côte d'Ivoire                   224599.01
Uganda                          213950.38
Malawi                          181098.71
Mozambique                      161407.98
Mauritania                      156665.46
Mali                            149928.33
Madagascar                      131197.73
Niger                           126707.58
Benin                           124771.22
Tunisia                         124167.20
Zambia                          103223.77
Burkina Faso                 

In [6]:
# Question 7 contd
df2.idxmax()

'Nigeria'

In [7]:
# Question 10
df3=df.corr()
df3

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

therefore the year with the least correlation with element code is 2016 with 0.023444 correlation

In [8]:
# Question 11
len(df['Area'].unique())

49

In [9]:
# Question 12
df[df['Element']=='Stock Variation'].groupby('Element').sum()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Element                                                                   
Stock Variation     571566   11329527      21464704  58749.83  34910.99   

                    Y2016     Y2017     Y2018  
Element                                        
Stock Variation  33140.12  54316.91  20577.91

therefore the year 2014 has the highest 'Stock Variation' with 58749.83

In [10]:
# Question 13
df[df['Item']=='Wine'].groupby('Item').sum()[['Y2015','Y2018']]


Y2015    Y2018
Item                  
Wine  4251.81  4039.32

In [11]:
# Question 14

#to select the 7th least country,the index location 'iloc' starts from '0', so we will use 6

df.groupby(['Area'])['Y2017'].sum().sort_values().reset_index().iloc[6]

Area     Guinea-Bissau
Y2017         19102.77
Name: 6, dtype: object

In [12]:
# Question 15

a=df['Y2014'].isnull().sum()
b=a/len(df['Y2014'])
c=round(b*100,3)
print('Total number: {}'.format(a))
print('Total percentage: {}%'.format(c))

Total number: 1589
Total percentage: 2.607%


In [13]:
# Question 19

print('Mean= {}'.format(round(df['Y2017'].mean(),2)))
print('Standard Deviation= {}'.format(round(df['Y2017'].std(),2)))

Mean= 140.92
Standard Deviation= 1671.86


In [14]:
# Question 20

df[df['Element']=='Processing'].groupby('Element').sum()[['Y2017']]

Y2017
Element             
Processing  292836.0